tests/run/i3006b.check

```diff
@@ -1,3 +1,3 @@
-Foo$$init$$$bar$1
-Foo$$init$$$bar$2
-Bar$$init$$$bar$1
+Foo$$_$bar$1
```

Comment:
Why was this changed? Isn't this going to make debugging more difficult?

Reply:
It makes things more regular. Fully qualified names do skip terms like this. The problem was that we treated the last separator of an expanded name as different from the others. This looks wrong. It's a different question what fully qualified names should do. AFAIK scalac skips term components completely.